WordPress/gutenberg
# Full Site Editing: Update the template parts admin list with new columns and views

## Description

We've recently updated the `wp_template` admin list with more columns and views to show all the available templates, with more informations (such as their corresponding theme, or if they were originally provided as files).

This PR adds the same features to the `wp_template_part`.

- Add support for excerpts (called "Description" in the list).
- Add columns for status and theme.
- Add a new view for auto-draft items.

To avoid duplicating a lot of code, or having hooks for one post type in the file registering the other, I've moved the callbacks functions in a separate utilities file.

## How has this been tested?

- With an FSE theme open `/wp-admin/edit.php?post_type=wp_template_part`.
- Make sure the new features show up as expected.
- Check the templates list (`/wp-admin/edit.php?post_type=wp_template`) for regressions.

## Screenshots 

<img width="1300" alt="Screenshot 2020-11-20 at 16 52 08" src="https://user-images.githubusercontent.com/2070010/99826805-bd67b300-2b50-11eb-8be1-5ca4c3f8aa98.png">

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
